packages/api/src/folder/service.ts

```diff
@@ -0,0 +1,124 @@
+import createError from "http-errors";
+import { logger } from "@stela/logger";
+import { db } from "../database";
+import type {
+  Folder,
+  FolderColumnsForUpdate,
+  FolderLink,
+  PatchFolderRequest,
+} from "./models";
+import { requestFieldsToDatabaseFields } from "./helper";
+import type { GetAccountArchiveResult } from "../account/models";
+import { accessService } from "../access/service";
+import { permission } from "../access/permission";
+
+export const getFolderById = async (requestQuery: {
+  folderId: string;
+}): Promise<Folder> => {
+  const folder = await db
+    .sql<Folder>("folder.queries.get_folder_by_id", {
+      folderId: requestQuery.folderId,
+    })
+    .catch((err) => {
+      logger.error(err);
+      throw new createError.InternalServerError("Failed to retrieve folder");
+    });
+
+  if (folder.rows[0] === undefined) {
+    throw new createError.NotFound("Folder not found.");
+  }
+
+  return folder.rows[0];
+};
+
+export const getFolderLinkByFolderId = async (requestQuery: {
+  folderId: string;
+}): Promise<FolderLink> => {
+  const folder = await db
+    .sql<FolderLink>("folder.queries.get_folder_link_by_folder_id", {
+      folderId: requestQuery.folderId,
+    })
+    .catch((err) => {
+      logger.error(err);
+      throw new createError.InternalServerError(
+        "Failed to retrieve folder link"
+      );
+    });
+
+  if (folder.rows[0] === undefined) {
+    throw new createError.NotFound("Folder link not found.");
+  }
+
+  return folder.rows[0];
+};
+
+function buildPatchQuery(patchFolderRequest: FolderColumnsForUpdate): string {
+  const updates = Object.entries(patchFolderRequest)
+    .filter(([key, value]) => value !== undefined && key !== "folderId")
+    .map(([key, _]) => `${key} = :${key}`);
+
+  if (updates.length === 0) {
+    throw new createError.BadRequest("Request cannot be empty");
+  }
+
+  const query = `
+    UPDATE folder
+    SET ${updates.join(", ")}
+    WHERE folderid = :folderId
+    RETURNING folder.folderid AS "folderId"
+  `;
+
+  return query.trim();
+}
+
+const validateCanPatchFolder = async (
```

Comment:
I'd consider combining the three queries this function uses into a single `get_folder_permissions_data`. Maybe that means this whole function moves to the access folder and that becomes general home for access checks like this.